### PR TITLE
Select position of ThermalAssistant

### PR DESCRIPTION
**Pull request by webbsimax**
Allow user to select position of ThermalAssistant.
Includes avoiding infoboxes if in the way.

Brief summary of the changes
----------------------------

Related issues and discussions
------------------------------

### CODE REVIEW

File: src/Dialogs/Settings/dlgConfiguration.cpp [hunk 1]
```diff
@@ -359,7 +359,7 @@ void dlgConfigurationShowModal()
                                       menu->Create(parent, rc, style);
                                       return menu;
                                     }));
-
+
```
Comment (MaxKellermann) [inline src/Dialogs/Settings/dlgConfiguration.cpp:362]: Whitespace change?

File: src/Dialogs/Settings/Panels/GaugesConfigPanel.cpp [hunk 2]
```diff
@@ -136,8 +159,9 @@ GaugesConfigPanel::Save(bool &_changed)
   changed |= SaveValue(AutoCloseFlarmDialog, ProfileKeys::AutoCloseFlarmDialog,
                        ui_settings.traffic.auto_close_dialog);
 
-  changed |= SaveValue(EnableTAGauge, ProfileKeys::EnableTAGauge,
-                       ui_settings.enable_thermal_assistant_gauge);
+  if (SaveValueEnum(TAPosition, ProfileKeys::TAPosition,
+                    ui_settings.thermal_assistant_position))
+    require_restart = changed = true;
```
Comment (MaxKellermann) [inline src/Dialogs/Settings/Panels/GaugesConfigPanel.cpp:164]: Does this require a restart?

File: src/MainWindow.cpp [hunk 3]
```diff
@@ -292,7 +292,29 @@ MainWindow::ReinitialiseLayoutTA(PixelRect rc,
 {
   unsigned sz = std::min(layout.control_size.cy,
                          layout.control_size.cx) * 2;
-  rc.right = rc.left + sz;
+
+  switch (CommonInterface::GetUISettings().thermal_assistant_position) {
+    case (UISettings::ThermalAssistantPosition::BOTTOM_LEFT_AVOID_IB):
```
Comment (MaxKellermann) [inline src/MainWindow.cpp:297]: Bad indent

File: src/MainWindow.cpp [hunk 4]
```diff
@@ -292,7 +292,29 @@ MainWindow::ReinitialiseLayoutTA(PixelRect rc,
...
+      rc.bottom = GetMainRect().bottom;
+      rc.right = GetMainRect().right;
+      rc.left = rc.right - sz;
+      break;
+    case (UISettings::ThermalAssistantPosition::BOTTOM_RIGHT):
+      rc.right = GetMainRect().right;
+      rc.left = rc.right - sz;
+      break;
+    default:
+      break;
```
Comment (MaxKellermann) [inline src/MainWindow.cpp:316]: Pointless `default` case.

File: src/Profile/UIProfile.cpp [hunk 5]
```diff
@@ -130,7 +130,7 @@ Profile::Load(const ProfileMap &map, UISettings &settings)
   if (settings.custom_dpi < 120 || settings.custom_dpi > 520)
     settings.custom_dpi = 0;
 
-  map.Get(ProfileKeys::EnableTAGauge, settings.enable_thermal_assistant_gauge);
+  map.GetEnum(ProfileKeys::TAPosition, settings.thermal_assistant_position);
```
Comment (MaxKellermann) [inline src/Profile/UIProfile.cpp:133]: This loses existing configuration. If you refactor settings, you should migrate old configuration data.

**Comment by MaxKellermann** [comment]
"Edit News.txt" is a bad commit message. Maybe fold this change into the main commit?

**Comment by webbsimax** [comment]
Created new PR #434

**Pull request closed without merging**